lib/active_graph/transactions.rb

```diff
@@ -28,6 +28,10 @@ def read_transaction(**config, &block)
 
       alias transaction write_transaction
 
+      def lock_node(node)
+        node.as(:n).query.remove('n._LOCK_').exec if tx&.open? || explicit_session&.open?
```

Comment:
it would be good to ensure that your specs fail with this code:
```suggestion
        node.as(:n).query.exec if tx&.open? || explicit_session&.open?
```

Reply:
Where is the tx boundary set in specs? What tx do new threads use?

Reply:
The specs are too complex to be confident without failing version.

Reply:
with disabling my code in `lock_node` I get following error so we now have a simpler specs with proper failure without fix.

```
  1) ActiveGraph::Relationship::Persistence::QueryFactory#build! non-subclassed concurrent update does not create duplicate has_one relationship
     Failure/Error: expect(ActiveGraph::Base.query("MATCH (node2:`ToClass`)<-[rel1:`HAS_REL`]-(from_class:`FromClass`) return from_class").to_a.size).to eq(1)
     
       expected: 1
            got: 19
     
       (compared using ==)
     # ./spec/e2e/relationship/persistence/query_factory_spec.rb:126:in `block (5 levels) in <top (required)>'
     ```

Reply:
@klobuczek every thread has unique tx, I checked their object_id.